src/components/Modal/Competency/AddCompetencyModelModal.js

```diff
@@ -61,28 +81,29 @@ const AddCompetencyModelModal = props => {
   const validation = useFormik({
     enableReinitialize: true,
     initialValues: {
+      domainId: '',
       code: '',
       name: '',
-      desc: '',
+      definition: '',
       proficiencyKeyActions: staticProficiencyKeyActions || [],
     },
     validationSchema: Yup.object({
       code: Yup.string().required('Please enter a code name'),
       name: Yup.string().required('Please enter a name'),
-      desc: Yup.string().required('Please enter a competency description'),
+      definition: Yup.string().required('Please enter a competency definition'),
       proficiencyKeyActions: Yup.array().of(
         Yup.object().shape({
-          // level: Yup.string().required('Proficiency level is required'),
           keyAction: Yup.string().required(
             'Proficiency key action is required'
           ),
         })
       ),
     }),
     onSubmit: (values, { resetForm }) => {
-      console.log(values)
-      //   dispatch(addCompetencyDetails(values))
-      //   resetForm()
+      const domainId = _id
+      const competencyModelDataSubmit = { ...values, domainId }
+      dispatch(addCompetencyDetails(competencyModelDataSubmit))
+      resetForm()
```

Comment:
Transfer resetForm to checker if the transaction is successful. 

On submission of the form and the transaction did not go through, the user will have to encode every detail in the form again.

Reply:
Transferred resetForm already to the checker if submission is successful.